tempesta_fw/http.c

```diff
@@ -660,24 +673,29 @@ tfw_http_req_zap_error(struct list_head *equeue)
 
 	list_for_each_entry_safe(req, tmp, equeue, fwd_list) {
 		list_del_init(&req->fwd_list);
-		switch(req->httperr.status) {
-		case 404:
-			tfw_http_send_404(req, req->httperr.reason);
-			break;
-		case 500:
-			tfw_http_send_500(req, req->httperr.reason);
-			break;
-		case 502:
-			tfw_http_send_502(req, req->httperr.reason);
-			break;
-		case 504:
-			tfw_http_send_504(req, req->httperr.reason);
-			break;
-		default:
-			TFW_WARN("Unexpected response error code: [%d]\n",
-				 req->httperr.status);
-			tfw_http_send_500(req, req->httperr.reason);
-			break;
+		if (!ret) {
```

Comment:
Why do we stop sending error responses after first failure? If there is not enough memory, then memory reclamation process is running and we can keep trying to do our job.

Reply:
> Why do we stop sending error responses after first failure? 

As I tried to explain in the comment above, once a response cannot be sent, it's irrelevant whether subsequent responses can be sent or not. The pairing is broken, and the connection must be closed for that reason. And by design any subsequent responses will not be sent to a client anyway.

Or, perhaps I didn't understand the question?

Reply:
@keshonok thank you for the review.

As I understand for example `tfw_http_conn_repair()`, `equeue` can contain requests from different client connections, so in the code we face problems on sending error messages to different client connections and the error handling for the connections must not depend on each other.

Reply:
> can contain requests from different client connections

Ah, yes, of course. Didn't understand it at first. You are correct, each erroneous request must be processed independently, and processing in this function must not stop after first error.